homeassistant/components/thermostat/homematic.py

```diff
@@ -29,20 +36,29 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     """Setup the Homematic thermostat."""
     devices = []
     try:
-        homegear = ServerProxy(config[CONF_ADDRESS])
         for name, device_cfg in config[CONF_DEVICES].items():
+            address = config[CONF_ADDRESS]
+            homegear = ServerProxy(address)
```

Comment:
This object is only used inside this try block. Why instantiate one per loop iteration as they are all the same address?